resultsdbupdater/utils.py

```diff
@@ -64,6 +64,54 @@ def create_result(log, testcase, outcome, ref_url, data, groups=None, note=None)
     post_req.raise_for_status()
 
 
+def update_common_ci_data(msg, outcome, data):
+    data.update(msg.contact_dict)
+    data['recipients'] = msg.recipients
+
+    update_publisher_id(data=data, msg=msg)
+
+    if outcome == 'ERROR':
+        data['error_reason'] = msg.error_reason[:256]
+
+        issue_url = msg.get('error', 'issue_url', default=None)
+        if issue_url:
+            data['issue_url'] = issue_url
+
+
+def product_scenario_builds(product_scenario, products):
+    builds_data = []
+    for product in products:
+        if product.get('type') != 'product-build':
+            continue
+
+        try:
+            build_data = {
+                'product_scenario': product_scenario,
+
+                # required values
+                'type': product['type'],
+                'item': product['id'],
+                'name': product['name'],
+                'version': product['version'],
+                'release': product['release'],
+                'artifacts': product['artifacts'],
```

Comment:
@msrb I just noticed that this is list of dicts. ResultsDB would serialize the items as strings (not even JSON). There is also some size limit introduced by the type of index in the postgres DB - resultsdb-updater currently crops large items (to 8k characters), but in this case it won't be possible.

Fix is to serialize each item as JSON but updater would need to drop large items or refuse to store the data.

Any ideas?

Reply:
So I found [this PR](https://pagure.io/fedora-ci/messages/pull-request/72) which (among other things) talks about the artifacts list. From the discussion there, it seems like the list of artifacts is super-important and it shouldn't be empty.

@hluk I don't really see into the implementation details of the resultsdb. What type of index is currently applied on the column where we store this data? Is it possible to use different type, of possibly do not have any index for this column at all? I am not sure what kind of queries are supported...

Reply:
I think it's not possible to use different type or indexing without changing the ResultsDB code (and probably writing a DB migration).

Reply:
That is fine, I don't think we need to fix this problem as part of this chage/pull request. Do you already have a RFE filed somewhere? Or should I open one?

Reply:
Can you please open one?

Reply:
EXD jira?

Reply:
I changed `artifacts` to be stored as list of JSON serialize objects.